CRM/Core/BAO/CustomGroup.php

```diff
@@ -654,7 +654,14 @@ protected static function validateSubTypeByEntity($entityType, $subType) {
     }
     $subTypes = CRM_Contact_BAO_ContactType::subTypeInfo($entityType, TRUE);
     $subTypes = array_merge($subTypes, CRM_Event_PseudoConstant::eventType());
-    if (!array_key_exists($subType, $subTypes)) {
+    // When you create a new contact type it gets saved in mixed case in the database.
+    // Eg. "Service User" becomes "Service_User" in civicrm_contact_type.name
+    // But that field does not differentiate case (eg. you can't add Service_User and service_user because mysql will report a duplicate error)
+    // webform_civicrm and some other integrations pass in the name as lowercase to API3 Contact.duplicatecheck
+    // Since we can't actually have two strings with different cases in the database perform a case-insensitive search here:
+    $subTypes = array_change_key_case($subTypes, CASE_LOWER);
+    if (!array_key_exists(mb_strtolower($subType), $subTypes)) {
```

Comment:
I'm having trouble evaluating this PR because it sounds like the external callers are doing something wrong. But in terms of evaluating it for normal usage if you were to pass in a proper subtype then it would do the same as before and so seems safe.

Minor technical note: Are labels being passed in here or names? Hopefully it's names, and I think it's recommended somewhere that names only contain ascii chars, and array_change_key_case seems to mangle non-ascii so the comparison would fail. In fact if I create a contact type of Доктор the name becomes `D9qVh1AO1KWxjcFj4bzVuR6ZxCAyoCPkv0LRsP747CdUn2sAhm9TlgYMCGUumJo`, which is maybe extreme but produces a "safe" name. All that to say the mb_strtolower() isn't necessarily wrong, just probably wouldn't work as intended if it were non-ascii.

Aside: Regardless of the PR, if you pass in 'Contact', 'Individual' to this function it says invalid, but that's maybe not its purpose - although it checks entity!=Contact so it must be expecting it. Anyway.